### PR TITLE
feat: Implement POST /register endpoint for user creation

### LINKED ISSUE #19: Add feature: Create new users (router, service, use case, repo implementation)
### Overview
Implement a feature to allow the creation of new users within the API. This includes changes across several layers of the project structure to ensure code modularity, reusability, and adherence to clean architecture principles.

### Requirements
- **Router**: Add new endpoints for user creation (e.g., `POST /users`).
- **Service Layer**: Implement service logic to process user creation requests and interact with use cases.
- **Use Case**: Define and implement the core business logic for user creation, including validation and orchestration between repository and service layers.
- **Repository Implementation**: Add methods to persist new users to the data store. Use dependency injection to keep the code scalable and testable.
- **Data Models/Schemas**: Update or add necessary Pydantic models for request/response validation.
- **Testing**: Add unit and integration tests for routers, services, and repository implementations.

### Acceptance Criteria
- New users can be created using an API endpoint (e.g., `POST /users`).
- Input validation handles edge cases (e.g., duplicate users, invalid data).
- Application of design patterns (e.g., Repository, Dependency Injection) and SOLID principles throughout the code.
- Code is well-organized and adheres to the existing architecture conventions in the project.
- Sufficient unit and integration tests are provided for all new components.

### Additional Notes
- Follow existing project conventions for dependency injection and service composition.
- Make the implementation modular and reusable for future extension (e.g., user roles or authentication integration).
- Document endpoint(s) and business logic where appropriate.

### PR DESCRIPTION
**Pull request by genesis-morales**
## 📋 Overview
Implement POST /register endpoint to allow user creation via the API, including password hashing and email validation.

Resolves #19 
---

### Application Layer
- Created `CreateUserUseCase` to handle user creation flow:
  - Normalizes email
  - Checks for duplicate users via existing `UserRepository`
  - Delegates password handling to the repository
- In`user_dto_mapper.py`:
  - `map_create_request_to_entity()` (DTO → Entity)
 
### Infrastructure Layer
- Updated `UserRepositoryImpl.create_user()` to:
  - Hash passwords using `bcrypt` before persisting
  - Check for duplicate email at DB level
- Updated `UserModel` / users table to add `password_hash`

### Presentation Layer
- Added `POST /register` endpoint:
  - Uses existing `UserService` via DI (`get_user_service`)
  - Returns `201` on success with `UserResponse`
  - Returns `409` on duplicate user
  - Returns `400` on business validation errors
---

### CODE REVIEW

File: src/app/features/infrastructure/repository/user_repository_impl.py [hunk 1]
```diff
@@ -83,3 +85,43 @@ async def update(self, entity: T) -> Optional[T]:
 
     async def delete(self, entity_id: ID) -> bool:
         pass
+
+    async def create_user(self, user: UserEntity, password: str) -> UserEntity:
```
Comment (Alonsovn) [inline src/app/features/infrastructure/repository/user_repository_impl.py]: there is already a function to save the entities in the database, that is define from the Base class.
Implement this logic in the function save() 
https://github.com/EndToEndLabCR/template-api-python/blob/c87aacd388be24b73b6ae2435226ef970a3973f6/src/app/features/infrastructure/repository/user_repository_impl.py#L72

**Review by Alonsovn — CHANGES_REQUESTED** [review]

**Review by Alonsovn — APPROVED** [review]
